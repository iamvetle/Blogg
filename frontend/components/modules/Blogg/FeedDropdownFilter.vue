<template>
	<div data-test="filter_tags" v-if="isReady">
		<!-- v-model changes the selected tags directly -->
		<USelectMenu
			:options="tags"
			v-model="searchStore.tagFilterPart"
			searchable
			searchablePlaceholder="Search a tag..."
			multiple
			placeholder="Option tags"
			size="lg"
			selectClass="items-center bg-primaryContainer text-onPrimaryContainer outline-none text-sm"
		>
			<template #leading>
				<Icon name="filter" size="1rem" />
			</template>

			<!-- When closed -->
			<template #label>
				<div class="text-md flex items-center" v-if="searchStore.tagFilterPart.length">
					<span class="break-words"
						>{{ searchStore.tagFilterPart.length }} filter</span
					>
				</div>
			</template>

			<!-- The options when open-->
			<template #option="{ option: tag }">
				<div class="text-sm">
					<span class="truncate">{{ tag }}</span>
				</div>
			</template>
		</USelectMenu>
	</div>
</template>

<script setup lang="ts">

const searchStore = useSearchStore();

/** State that holds all the tags - fetched */
const tags = ref(<string[]>[]);

/** Watches how to */

onMounted(async () => {
	console.debug("is this printet")
	const response = await getAllTags();

	if (response) {
		console.debug("is this printet ")
		for (let tag of response) {
			console.debug(tag)
			tags.value.push(tag.name);
		}
	}
});
// useWatchFeedDropdownFilter();


/**
 * ? maybe I should have a "isReady" for most components?
 * 
 * ! through an error if I shorten it to just => sea..
 * ! TO SELF: to not use ? with .value, that is reactive referances
 */

// THIS under, also works
// const isReady = computed(() => {
// 	if (searchStore.tagFilterPart.value) {
// 		return true
// 	} else {
// 		return false
// 	}
// })

const isReady = computed(() => searchStore.tagFilterPart ? true : false)
</script>

<style scoped></style>
