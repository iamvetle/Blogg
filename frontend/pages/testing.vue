<template>
  <div class="flex">

    <form @submit.prevent method="post" class="flex flex-col" enctype="multipart/form-data">

      <label for="name">Name:</label>
      <input type="text" name="name" id="name">

      <label for="email">Email</label>
      <input type="email" name="email" id="email">

      <label for="message">Message:</label>
      <textarea name="message" id="" cols="30" rows="10" placeholder="Write a message here"></textarea>

      <label for="file">File:</label>
      <input type="file" name="file" id="file">

      <button type="submit">Submit</button>
    </form>

  </div>
</template>

<script setup lang="ts">
onMounted(() => {

  const form = document.querySelector("form")

  form?.addEventListener("submit", (e: any) => {


    const formData = new FormData(form)

    for (let item of formData) {
      console.log(item[0], item[1])
    }
    
    // ta axios of prÃ¸v fetch her
  })





})

// watchEffect(() => {

// })

</script>

<style scoped></style>