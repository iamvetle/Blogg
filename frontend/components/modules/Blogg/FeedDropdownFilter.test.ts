import FeedDropdownFilter from './FeedDropdownFilter.vue'
import { VueWrapper, shallowMount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

let wrapper: VueWrapper;
let pinia: any = createTestingPinia();

const factory = () => {
    return shallowMount(FeedDropdownFilter, {
        global: {
            plugins: [pinia],
            components: {},
            mocks: {},
            stubs: {
                FilterBox:true
            },
        },
        props: {},
        slots: {}
    })
};

describe('Testing FeedDropdownFilter', () => {

    beforeEach(() => {
    });

    afterEach(() => {
        if (wrapper) {
            wrapper.unmount();
        }
    });

    test('Should exist', () => {
      wrapper = factory()

      expect(wrapper.exists()).toBe(true)
    })
    test('Should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })


});