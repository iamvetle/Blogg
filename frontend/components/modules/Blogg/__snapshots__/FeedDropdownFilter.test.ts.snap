// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing FeedDropdownFilter > Should match snapshot 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    />,
    "_context": {
      "app": [Circular],
      "components": {
        "FilterBox": {
          "name": "FilterBox",
        },
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {
          "$pinia": {
            "_a": [Circular],
            "_e": EffectScope {
              "_active": true,
              "cleanups": [],
              "detached": true,
              "effects": [],
              "parent": undefined,
              "scopes": [
                EffectScope {
                  "_active": true,
                  "cleanups": [],
                  "detached": false,
                  "effects": [],
                  "index": 0,
                  "parent": [Circular],
                },
                EffectScope {
                  "_active": true,
                  "cleanups": [],
                  "detached": false,
                  "effects": [],
                  "index": 1,
                  "parent": [Circular],
                },
                EffectScope {
                  "_active": true,
                  "cleanups": [],
                  "detached": false,
                  "effects": [],
                  "index": 2,
                  "parent": [Circular],
                },
              ],
            },
            "_p": [
              [Function],
              [Function],
              [Function],
            ],
            "_s": Map {
              "search" => {
                "$dispose": [Function],
                "$id": "search",
                "$onAction": [Function],
                "$patch": [MockFunction $patch],
                "$reset": [MockFunction spy],
                "$subscribe": [Function],
                "_customProperties": Set {},
                "_hmrPayload": {
                  "actions": {
                    "resetStore": [Function],
                  },
                  "getters": {},
                  "hotState": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": {},
                    "_value": {},
                    "dep": undefined,
                  },
                  "state": [
                    "searchPart",
                    "categoryFilterPart",
                    "tagFilterPart",
                  ],
                },
                "_hotUpdate": [Function],
                "_p": [Circular],
                "categoryFilterPart": null,
                "resetStore": [MockFunction spy],
                "searchPart": null,
                "tagFilterPart": null,
              },
              "Store for containing posts and related to posts" => {
                "$dispose": [Function],
                "$id": "Store for containing posts and related to posts",
                "$onAction": [Function],
                "$patch": [MockFunction $patch],
                "$reset": [MockFunction spy],
                "$subscribe": [Function],
                "_customProperties": Set {},
                "_hmrPayload": {
                  "actions": {
                    "resetStore": [Function],
                  },
                  "getters": {},
                  "hotState": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": {},
                    "_value": {},
                    "dep": undefined,
                  },
                  "state": [
                    "posts",
                    "allComments",
                    "followingPosts",
                    "allCategories",
                    "allTags",
                    "loggedInUserPosts",
                    "baseFetchURL",
                    "baseLoggedInUserPostsURL",
                  ],
                },
                "_hotUpdate": [Function],
                "_p": [Circular],
                "allCategories": null,
                "allComments": null,
                "allTags": null,
                "baseFetchURL": "http://localhost:8888/api/feed/",
                "baseLoggedInUserPostsURL": "http://localhost:8888/api/min-side/posts/",
                "followingPosts": null,
                "loggedInUserPosts": null,
                "posts": null,
                "resetStore": [MockFunction spy],
              },
              "Pagination Store" => {
                "$dispose": [Function],
                "$id": "Pagination Store",
                "$onAction": [Function],
                "$patch": [MockFunction $patch],
                "$reset": [MockFunction spy],
                "$subscribe": [Function],
                "_customProperties": Set {},
                "_hmrPayload": {
                  "actions": {
                    "resetStore": [Function],
                  },
                  "getters": {},
                  "hotState": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": {},
                    "_value": {},
                    "dep": undefined,
                  },
                  "state": [
                    "activeFetchURL",
                    "next_page",
                    "previous_page",
                    "last_page_link",
                    "all_pages_count",
                    "number_of_posts",
                    "current_page_number",
                  ],
                },
                "_hotUpdate": [Function],
                "_p": [Circular],
                "activeFetchURL": "http://localhost:8888/api/feed/",
                "all_pages_count": 0,
                "current_page_number": 0,
                "last_page_link": "",
                "next_page": "",
                "number_of_posts": 0,
                "previous_page": "",
                "resetStore": [MockFunction spy],
              },
            },
            "_testing": true,
            "app": [Circular],
            "install": [Function],
            "state": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": {
                "Pagination Store": {
                  "activeFetchURL": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": "http://localhost:8888/api/feed/",
                    "_value": "http://localhost:8888/api/feed/",
                    "dep": undefined,
                  },
                  "all_pages_count": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": 0,
                    "_value": 0,
                    "dep": undefined,
                  },
                  "current_page_number": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": 0,
                    "_value": 0,
                    "dep": undefined,
                  },
                  "last_page_link": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": "",
                    "_value": "",
                    "dep": undefined,
                  },
                  "next_page": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": "",
                    "_value": "",
                    "dep": undefined,
                  },
                  "number_of_posts": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": 0,
                    "_value": 0,
                    "dep": undefined,
                  },
                  "previous_page": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": "",
                    "_value": "",
                    "dep": undefined,
                  },
                },
                "Store for containing posts and related to posts": {
                  "allCategories": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": null,
                    "_value": null,
                    "dep": undefined,
                  },
                  "allComments": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": null,
                    "_value": null,
                    "dep": undefined,
                  },
                  "allTags": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": null,
                    "_value": null,
                    "dep": Set {},
                  },
                  "baseFetchURL": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": "http://localhost:8888/api/feed/",
                    "_value": "http://localhost:8888/api/feed/",
                    "dep": undefined,
                  },
                  "baseLoggedInUserPostsURL": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": "http://localhost:8888/api/min-side/posts/",
                    "_value": "http://localhost:8888/api/min-side/posts/",
                    "dep": undefined,
                  },
                  "followingPosts": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": null,
                    "_value": null,
                    "dep": undefined,
                  },
                  "loggedInUserPosts": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": null,
                    "_value": null,
                    "dep": undefined,
                  },
                  "posts": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": null,
                    "_value": null,
                    "dep": undefined,
                  },
                },
                "search": {
                  "categoryFilterPart": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": null,
                    "_value": null,
                    "dep": undefined,
                  },
                  "searchPart": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": null,
                    "_value": null,
                    "dep": undefined,
                  },
                  "tagFilterPart": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": null,
                    "_value": null,
                    "dep": undefined,
                  },
                },
              },
              "_value": {
                "Pagination Store": {
                  "activeFetchURL": "http://localhost:8888/api/feed/",
                  "all_pages_count": 0,
                  "current_page_number": 0,
                  "last_page_link": "",
                  "next_page": "",
                  "number_of_posts": 0,
                  "previous_page": "",
                },
                "Store for containing posts and related to posts": {
                  "allCategories": null,
                  "allComments": null,
                  "allTags": null,
                  "baseFetchURL": "http://localhost:8888/api/feed/",
                  "baseLoggedInUserPostsURL": "http://localhost:8888/api/min-side/posts/",
                  "followingPosts": null,
                  "loggedInUserPosts": null,
                  "posts": null,
                },
                "search": {
                  "categoryFilterPart": null,
                  "searchPart": null,
                  "tagFilterPart": null,
                },
              },
              "dep": undefined,
            },
            "use": [Function],
          },
        },
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {
        Symbol(pinia): {
          "_a": [Circular],
          "_e": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [],
            "parent": undefined,
            "scopes": [
              EffectScope {
                "_active": true,
                "cleanups": [],
                "detached": false,
                "effects": [],
                "index": 0,
                "parent": [Circular],
              },
              EffectScope {
                "_active": true,
                "cleanups": [],
                "detached": false,
                "effects": [],
                "index": 1,
                "parent": [Circular],
              },
              EffectScope {
                "_active": true,
                "cleanups": [],
                "detached": false,
                "effects": [],
                "index": 2,
                "parent": [Circular],
              },
            ],
          },
          "_p": [
            [Function],
            [Function],
            [Function],
          ],
          "_s": Map {
            "search" => {
              "$dispose": [Function],
              "$id": "search",
              "$onAction": [Function],
              "$patch": [MockFunction $patch],
              "$reset": [MockFunction spy],
              "$subscribe": [Function],
              "_customProperties": Set {},
              "_hmrPayload": {
                "actions": {
                  "resetStore": [Function],
                },
                "getters": {},
                "hotState": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": {},
                  "_value": {},
                  "dep": undefined,
                },
                "state": [
                  "searchPart",
                  "categoryFilterPart",
                  "tagFilterPart",
                ],
              },
              "_hotUpdate": [Function],
              "_p": [Circular],
              "categoryFilterPart": null,
              "resetStore": [MockFunction spy],
              "searchPart": null,
              "tagFilterPart": null,
            },
            "Store for containing posts and related to posts" => {
              "$dispose": [Function],
              "$id": "Store for containing posts and related to posts",
              "$onAction": [Function],
              "$patch": [MockFunction $patch],
              "$reset": [MockFunction spy],
              "$subscribe": [Function],
              "_customProperties": Set {},
              "_hmrPayload": {
                "actions": {
                  "resetStore": [Function],
                },
                "getters": {},
                "hotState": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": {},
                  "_value": {},
                  "dep": undefined,
                },
                "state": [
                  "posts",
                  "allComments",
                  "followingPosts",
                  "allCategories",
                  "allTags",
                  "loggedInUserPosts",
                  "baseFetchURL",
                  "baseLoggedInUserPostsURL",
                ],
              },
              "_hotUpdate": [Function],
              "_p": [Circular],
              "allCategories": null,
              "allComments": null,
              "allTags": null,
              "baseFetchURL": "http://localhost:8888/api/feed/",
              "baseLoggedInUserPostsURL": "http://localhost:8888/api/min-side/posts/",
              "followingPosts": null,
              "loggedInUserPosts": null,
              "posts": null,
              "resetStore": [MockFunction spy],
            },
            "Pagination Store" => {
              "$dispose": [Function],
              "$id": "Pagination Store",
              "$onAction": [Function],
              "$patch": [MockFunction $patch],
              "$reset": [MockFunction spy],
              "$subscribe": [Function],
              "_customProperties": Set {},
              "_hmrPayload": {
                "actions": {
                  "resetStore": [Function],
                },
                "getters": {},
                "hotState": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": {},
                  "_value": {},
                  "dep": undefined,
                },
                "state": [
                  "activeFetchURL",
                  "next_page",
                  "previous_page",
                  "last_page_link",
                  "all_pages_count",
                  "number_of_posts",
                  "current_page_number",
                ],
              },
              "_hotUpdate": [Function],
              "_p": [Circular],
              "activeFetchURL": "http://localhost:8888/api/feed/",
              "all_pages_count": 0,
              "current_page_number": 0,
              "last_page_link": "",
              "next_page": "",
              "number_of_posts": 0,
              "previous_page": "",
              "resetStore": [MockFunction spy],
            },
          },
          "_testing": true,
          "app": [Circular],
          "install": [Function],
          "state": RefImpl {
            "__v_isRef": true,
            "__v_isShallow": false,
            "_rawValue": {
              "Pagination Store": {
                "activeFetchURL": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "http://localhost:8888/api/feed/",
                  "_value": "http://localhost:8888/api/feed/",
                  "dep": undefined,
                },
                "all_pages_count": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": 0,
                  "_value": 0,
                  "dep": undefined,
                },
                "current_page_number": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": 0,
                  "_value": 0,
                  "dep": undefined,
                },
                "last_page_link": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "",
                  "_value": "",
                  "dep": undefined,
                },
                "next_page": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "",
                  "_value": "",
                  "dep": undefined,
                },
                "number_of_posts": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": 0,
                  "_value": 0,
                  "dep": undefined,
                },
                "previous_page": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "",
                  "_value": "",
                  "dep": undefined,
                },
              },
              "Store for containing posts and related to posts": {
                "allCategories": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "allComments": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "allTags": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": Set {},
                },
                "baseFetchURL": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "http://localhost:8888/api/feed/",
                  "_value": "http://localhost:8888/api/feed/",
                  "dep": undefined,
                },
                "baseLoggedInUserPostsURL": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "http://localhost:8888/api/min-side/posts/",
                  "_value": "http://localhost:8888/api/min-side/posts/",
                  "dep": undefined,
                },
                "followingPosts": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "loggedInUserPosts": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "posts": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
              },
              "search": {
                "categoryFilterPart": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "searchPart": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "tagFilterPart": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
              },
            },
            "_value": {
              "Pagination Store": {
                "activeFetchURL": "http://localhost:8888/api/feed/",
                "all_pages_count": 0,
                "current_page_number": 0,
                "last_page_link": "",
                "next_page": "",
                "number_of_posts": 0,
                "previous_page": "",
              },
              "Store for containing posts and related to posts": {
                "allCategories": null,
                "allComments": null,
                "allTags": null,
                "baseFetchURL": "http://localhost:8888/api/feed/",
                "baseLoggedInUserPostsURL": "http://localhost:8888/api/min-side/posts/",
                "followingPosts": null,
                "loggedInUserPosts": null,
                "posts": null,
              },
              "search": {
                "categoryFilterPart": null,
                "searchPart": null,
                "tagFilterPart": null,
              },
            },
            "dep": undefined,
          },
          "use": [Function],
        },
      },
      "reload": [Function],
    },
    "_instance": null,
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {
        "$pinia": {
          "_a": [Circular],
          "_e": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [],
            "parent": undefined,
            "scopes": [
              EffectScope {
                "_active": true,
                "cleanups": [],
                "detached": false,
                "effects": [],
                "index": 0,
                "parent": [Circular],
              },
              EffectScope {
                "_active": true,
                "cleanups": [],
                "detached": false,
                "effects": [],
                "index": 1,
                "parent": [Circular],
              },
              EffectScope {
                "_active": true,
                "cleanups": [],
                "detached": false,
                "effects": [],
                "index": 2,
                "parent": [Circular],
              },
            ],
          },
          "_p": [
            [Function],
            [Function],
            [Function],
          ],
          "_s": Map {
            "search" => {
              "$dispose": [Function],
              "$id": "search",
              "$onAction": [Function],
              "$patch": [MockFunction $patch],
              "$reset": [MockFunction spy],
              "$subscribe": [Function],
              "_customProperties": Set {},
              "_hmrPayload": {
                "actions": {
                  "resetStore": [Function],
                },
                "getters": {},
                "hotState": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": {},
                  "_value": {},
                  "dep": undefined,
                },
                "state": [
                  "searchPart",
                  "categoryFilterPart",
                  "tagFilterPart",
                ],
              },
              "_hotUpdate": [Function],
              "_p": [Circular],
              "categoryFilterPart": null,
              "resetStore": [MockFunction spy],
              "searchPart": null,
              "tagFilterPart": null,
            },
            "Store for containing posts and related to posts" => {
              "$dispose": [Function],
              "$id": "Store for containing posts and related to posts",
              "$onAction": [Function],
              "$patch": [MockFunction $patch],
              "$reset": [MockFunction spy],
              "$subscribe": [Function],
              "_customProperties": Set {},
              "_hmrPayload": {
                "actions": {
                  "resetStore": [Function],
                },
                "getters": {},
                "hotState": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": {},
                  "_value": {},
                  "dep": undefined,
                },
                "state": [
                  "posts",
                  "allComments",
                  "followingPosts",
                  "allCategories",
                  "allTags",
                  "loggedInUserPosts",
                  "baseFetchURL",
                  "baseLoggedInUserPostsURL",
                ],
              },
              "_hotUpdate": [Function],
              "_p": [Circular],
              "allCategories": null,
              "allComments": null,
              "allTags": null,
              "baseFetchURL": "http://localhost:8888/api/feed/",
              "baseLoggedInUserPostsURL": "http://localhost:8888/api/min-side/posts/",
              "followingPosts": null,
              "loggedInUserPosts": null,
              "posts": null,
              "resetStore": [MockFunction spy],
            },
            "Pagination Store" => {
              "$dispose": [Function],
              "$id": "Pagination Store",
              "$onAction": [Function],
              "$patch": [MockFunction $patch],
              "$reset": [MockFunction spy],
              "$subscribe": [Function],
              "_customProperties": Set {},
              "_hmrPayload": {
                "actions": {
                  "resetStore": [Function],
                },
                "getters": {},
                "hotState": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": {},
                  "_value": {},
                  "dep": undefined,
                },
                "state": [
                  "activeFetchURL",
                  "next_page",
                  "previous_page",
                  "last_page_link",
                  "all_pages_count",
                  "number_of_posts",
                  "current_page_number",
                ],
              },
              "_hotUpdate": [Function],
              "_p": [Circular],
              "activeFetchURL": "http://localhost:8888/api/feed/",
              "all_pages_count": 0,
              "current_page_number": 0,
              "last_page_link": "",
              "next_page": "",
              "number_of_posts": 0,
              "previous_page": "",
              "resetStore": [MockFunction spy],
            },
          },
          "_testing": true,
          "app": [Circular],
          "install": [Function],
          "state": RefImpl {
            "__v_isRef": true,
            "__v_isShallow": false,
            "_rawValue": {
              "Pagination Store": {
                "activeFetchURL": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "http://localhost:8888/api/feed/",
                  "_value": "http://localhost:8888/api/feed/",
                  "dep": undefined,
                },
                "all_pages_count": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": 0,
                  "_value": 0,
                  "dep": undefined,
                },
                "current_page_number": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": 0,
                  "_value": 0,
                  "dep": undefined,
                },
                "last_page_link": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "",
                  "_value": "",
                  "dep": undefined,
                },
                "next_page": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "",
                  "_value": "",
                  "dep": undefined,
                },
                "number_of_posts": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": 0,
                  "_value": 0,
                  "dep": undefined,
                },
                "previous_page": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "",
                  "_value": "",
                  "dep": undefined,
                },
              },
              "Store for containing posts and related to posts": {
                "allCategories": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "allComments": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "allTags": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": Set {},
                },
                "baseFetchURL": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "http://localhost:8888/api/feed/",
                  "_value": "http://localhost:8888/api/feed/",
                  "dep": undefined,
                },
                "baseLoggedInUserPostsURL": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": "http://localhost:8888/api/min-side/posts/",
                  "_value": "http://localhost:8888/api/min-side/posts/",
                  "dep": undefined,
                },
                "followingPosts": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "loggedInUserPosts": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "posts": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
              },
              "search": {
                "categoryFilterPart": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "searchPart": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
                "tagFilterPart": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": null,
                  "_value": null,
                  "dep": undefined,
                },
              },
            },
            "_value": {
              "Pagination Store": {
                "activeFetchURL": "http://localhost:8888/api/feed/",
                "all_pages_count": 0,
                "current_page_number": 0,
                "last_page_link": "",
                "next_page": "",
                "number_of_posts": 0,
                "previous_page": "",
              },
              "Store for containing posts and related to posts": {
                "allCategories": null,
                "allComments": null,
                "allTags": null,
                "baseFetchURL": "http://localhost:8888/api/feed/",
                "baseLoggedInUserPostsURL": "http://localhost:8888/api/min-side/posts/",
                "followingPosts": null,
                "loggedInUserPosts": null,
                "posts": null,
              },
              "search": {
                "categoryFilterPart": null,
                "searchPart": null,
                "tagFilterPart": null,
              },
            },
            "dep": undefined,
          },
          "use": [Function],
        },
      },
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.7",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": null,
  },
  "wrapperElement": <div>
    <span
      class="mb-2 w-full flex items-center text-center justify-center"
    >
      <button
        class="w-full h-full text-lg hover:text-primaryFixedDim rounded-md px-1 py-1 text-onPrimary flex text-center items-center justify-center"
        data-test="dropdown-button"
      >
         Filter posts 
      </button>
    </span>
    <div
      class="px-2 py-2"
      data-test="filter-component"
      style="display: none;"
    >
      <!-- The filter dropdown compontent-->
      <keep-alive-stub />
    </div>
  </div>,
}
`;
